apis/s3control/v1alpha1/zz_access_point_policy.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+import (
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// AccessPointPolicyParameters defines the desired state of AccessPointPolicy
+type AccessPointPolicyParameters struct {
+	// Region is which region the AccessPointPolicy will be created.
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+	// The Amazon Web Services account ID for owner of the bucket associated with
+	// the specified access point.
+	// +kubebuilder:validation:Required
+	AccountID *string `json:"accountID"`
+	// The policy that you want to apply to the specified access point. For more
+	// information about access point policies, see Managing data access with Amazon
+	// S3 access points (https://docs.aws.amazon.com/AmazonS3/latest/userguide/access-points.html)
+	// in the Amazon S3 User Guide.
+	// +kubebuilder:validation:Required
+	Policy                            *string `json:"policy"`
```

Comment:
Since there can only be one policy per AP it would be better to integrate the policy field into `AccessPoint` so users don' have to create two MRs. Much like it was for `Bucket` and `BucketPolicy` #1685.

It would also be good to provide the same structured schema for policies instead of passing a raw string.